### PR TITLE
Manually register the port mapping for k8s API

### LINKED ISSUE #6831: Rancher desktop WSL proxy no longer listens on 6443
As part of the [work](https://github.com/rancher-sandbox/rancher-desktop-networking/pull/21) to make wsl-proxy to dynamically receive portMapping we have mistakenly removed the static proxy that was listening for K8sAPI port 6443. 

To tackle this issue we can emit a portmapping event from the guest agent for port 6443 to wsl-proxy.

### PR DESCRIPTION
**Pull request by Nino-K**
This will send a port mapping event to Rancher-Desktop wsl-proxy for port 6443. This manual registration of k8s API port only applies when kubernetes along with wsl-integration are enabled.

Fixes: https://github.com/rancher-sandbox/rancher-desktop/issues/6831

### CODE REVIEW

File: cmd/rancher-desktop-guestagent/main.go [hunk 1]
```diff
@@ -143,6 +146,32 @@ func main() {
 	} else {
 		forwarder := forwarder.NewWSLProxyForwarder("/run/wsl-proxy.sock")
 		portTracker = tracker.NewAPITracker(forwarder, tracker.GatewayBaseURL, *adminInstall)
+		// manually register the port for K8s API, we would
+		// only want to send this manual portmapping under the
+		// following conditions:
```
Comment (mook-as) [inline cmd/rancher-desktop-guestagent/main.go]: It's unclear if this is describe a set of conditions that must _all_ be reached, or if we want _any_ condition to be reached.  Please rephrase.

File: cmd/rancher-desktop-guestagent/main.go [hunk 2]
```diff
@@ -181,7 +210,7 @@ func main() {
 			k8sServiceListenerIP := net.ParseIP(*k8sServiceListenerAddr)
 
 			if k8sServiceListenerIP == nil || !(k8sServiceListenerIP.Equal(net.IPv4zero) ||
-				k8sServiceListenerIP.Equal(net.IPv4(127, 0, 0, 1))) { //nolint:gomnd // IPv4 addr localhost
+				k8sServiceListenerIP.Equal(net.IPv4(127, 0, 0, 1))) { //nolint:gomnd,mnd // IPv4 addr localhost
```
Comment (mook-as) [inline cmd/rancher-desktop-guestagent/main.go]: See other PR: whitelisting `net.IPv4` seems like a better fix.

**Review by mook-as — CHANGES_REQUESTED** [review]
I'm not entirely happy that this logic is going in `-agent` instead of the JS side, especially since we still don't have any documentation on what feature is active/working in which circumstance.

**Comment by Nino-K** [comment]
> I'm not entirely happy that this logic is going in `-agent` instead of the JS side, especially since we still don't have any documentation on what feature is active/working in which circumstance.

I disagree, I think this logic does belong in the agent and the JS should only provide the configuration for this feature. However, it is worth noting that currently there is a disconnect in [where we forward](https://github.com/rancher-sandbox/rancher-desktop/blob/6bd509de6a4a3d5059c6df0b487c1c229236a4ec/pkg/rancher-desktop/backend/wsl.ts#L123) the port to the host for K8s API, and where we emit a port mapping event for wsl-integration. Given that these two events (port forwarding for host and wsl-distro for k8s API) are independent of each other, however,  I agree with the fact that this should be documented to elaborate this behavior further. I will create a follow-up PR to explain this in the README docs.

**Review by mook-as — APPROVED** [review]

**Pull request merged**